clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10619,36 +10607,66 @@ static void getOtherSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
 }
 
 // Add any sycl-post-link options that rely on a specific Triple.
-static void
-getTripleBasedSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
-                               const llvm::opt::ArgList &TCArgs,
-                               llvm::Triple Triple, ArgStringList &PostLinkArgs,
-                               bool SpecConsts, types::ID OutputType) {
+static void getTripleBasedSYCLPostLinkOpts(
+    const ToolChain &TC, const JobAction &JA, const llvm::opt::ArgList &TCArgs,
+    llvm::Triple Triple, ArgStringList &PostLinkArgs, bool SpecConsts,
+    types::ID OutputType, bool NewOffloadDriver = false) {
+  // Note: Do not use Triple when NewOffloadDriver is 'true'.
+  if (!NewOffloadDriver && (OutputType == types::TY_LLVM_BC)) {
```

Comment:
Some of the things here don't seem triple based `(if SpecConsts)`, `OutputType`. Is there a better place for them or are the variable names just a bit misleading?

Reply:
SpecConsts and OutputType depend on the target triple. For example, OutputType is computed as follows:
types::ID OutputType = **TargetTriple.isSPIROrSPIRV()** || IsSYCLNativeCPU
                               ? types::TY_Tempfiletable
                               : types::TY_LLVM_BC;